### PR TITLE
Add calculator functionality

### PR DESCRIPTION
**Pull request by ahmed-alfarouq**
What I did:
- Add `logic` directory and `calculate.js` & `operate.js` files.
- Add data in `app.js`
- Add `updateData` function

### CODE REVIEW

File: src/components/Calculator.jsx [hunk 1]
```diff
@@ -1,37 +1,44 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
Comment (eliasecasta) [inline src/components/Calculator.jsx:4]: - [ ] The Calculator component should import the calculate.js file from the logic directory, please change the import from the index to this file and refactor accordingly to comply with the project requirements.

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @ahmed-al-farouq, I am Elias, your reviewer for this PR

Well done so far with your project! All your calculator functionalities are working as expected and the calculator design is on point, keep it up!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [ ] The calculate import should happen inside the Calculator component, please make this small change in order to comply with the project requirements, everything else is great!

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Calculator.jsx (hunk 1 above)
Comment (ahmed-alfarouq) [reply src/components/Calculator.jsx:4]: That's right, I will do that.
Thanks for your the feedback.

Comment (henatan99) [reply src/components/Calculator.jsx:4]: This issue is well addressed. Great job! :clap: :+1:

**Review by henatan99 — APPROVED** [review]
Hi @ahmed-al-farouq , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**